### PR TITLE
sdk/logical: handle empty token type string values as TokenTypeDefault

### LINKED ISSUE #7270: Error 500: unknown token type "\"\"" when working with AppRole
**Describe the bug**

After upgrading to Vault 1.2.1, we can no longer login (or even read the configuration of) one of our AppRole roles. Other roles appear unaffected. We receive the following error:

Error reading auth/approle/role/my-role: Error making API request.

```
URL: GET https://vault.example.com:8200/v1/auth/approle/role/my-role
Code: 500. Errors:

* 1 error occurred:
        * unknown token type "\"\""
```

We've tried to re-write the AppRole with a valid configuration, but get:

```
Error writing data to auth/approle/role/my-role: Error making API request.

URL: PUT https://vault.example.com:8200/v1/auth/approle/role/my-role
Code: 500. Errors:

* 1 error occurred:
        * internal error
```

**To Reproduce**

1. Run `vault read auth/approle/role/my-role`
2. See error

**Expected behavior**

Shows configuration for that approle / allows login via approle.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.2.1
* Vault CLI Version (retrieve with `vault version`): 1.2.1
* Server Operating System/Architecture: RHEL 7.7 x86_64

Vault server configuration file(s):

```hcl
storage "consul" {
  address = "127.0.0.1:8500"
  path    = "secrets/"
}

ui = true
cluster_name = "cluster-name"
pid_file = "/var/run/vault.pid"
disable_clustering = true

listener "tcp" {
  address     = "1.2.3.5:8200"
  tls_cert_file = "/etc/pki/tls/certs/vault.crt"
  tls_key_file = "/etc/pki/tls/private/vault.key"
  tls_min_version = "tls12"
  proxy_protocol_behavior = "allow_authorized"
  proxy_protocol_authorized_addrs = "1.2.3.4,2.3.4.5,3.4.5.6"
  tls_disable_client_certs = "true"
}

api_addr = "https://vault-02.example.com:8200"
```

**Additional context**
We upgraded vault from 1.1.3 to 1.2.0 and then 1.2.1. We're not sure if this issue was present as well in 1.2.0 because of the other issues preventing AppRoles from working.

We have a number of other AppRoles, and this one seems to be the only one affected. Nothing really special about it's configuration:

```
{
  "bind_secret_id": true,
  "bound_cidr_list": null,
  "local_secret_ids": false,
  "period": 0,
  "policies": [
    "a-policy",
    "b-policy"
  ],
  "secret_id_bound_cidrs": null,
  "secret_id_num_uses": 0,
  "secret_id_ttl": 0,
  "token_bound_cidrs": null,
  "token_max_ttl": 600,
  "token_num_uses": 0,
  "token_ttl": 600
}
```

### PR DESCRIPTION
**Pull request by calvn**
Fixes #7270

### CODE REVIEW

File: builtin/credential/approle/path_role_test.go [hunk 1]
```diff
@@ -1843,6 +1844,23 @@ func createRole(t *testing.T, b *backend, s logical.Storage, roleName, policies
 // TestAppRole_TokenutilUpgrade ensures that when we read values out that are
 // values with upgrade logic we see the correct struct entries populated
 func TestAppRole_TokenutilUpgrade(t *testing.T) {
+	tests := []struct {
```
Comment (kalafut) [inline builtin/credential/approle/path_role_test.go]: Would be good to also have the case where there is no token_type key at all in the JSON, which is what we’d see coming from a pre-1.0 state.

**Comment by kalafut** [comment]
One test suggestion, otherwise LGTM.

**Comment by sivanag1974** [comment]
Even in vault 1.2.3, I am getting the error

./vault read -tls-skip-verify /auth/approle/role/adds_automation
Error reading auth/approle/role/adds_automation: Error making API request.

URL: GET https://10.65.226.28:46738/v1/auth/approle/role/adds_automation
Code: 500. Errors:

* 1 error occurred:
        * unknown token type "\"\""


[root@5a2ee033ssb8 tmp]# ./vault -v
Vault v1.2.3

**Comment by calvn** [comment]
@sivanag1974 Can you do a request against seal-status to check on the version of the server like so?

```
$ curl http://<your-vault-address>/v1/sys/seal-status
```

**Pull request merged**